src/config/validator/addr.rs

```diff
@@ -0,0 +1,75 @@
+//! Validator addresses (`tcp://` or `unix://`)
+
+use serde::{de::Error as DeError, Deserialize, Deserializer};
+use std::{
+    fmt::{self, Display},
+    path::PathBuf,
+    str::{self, FromStr},
+};
+
+use error::{KmsError, KmsErrorKind::*};
+
+#[derive(Clone, Debug)]
+pub enum ValidatorAddr {
```

Comment:
This type is arguably useful as an address for more things than just validators, and could potentially go in the `tendermint` crate.